packages/xo-web-6/src/App/Visualization/hostStats.tsx

```diff
@@ -0,0 +1,556 @@
+const xoCall = (method:string, params:object) => signedIn.then(() => xo.call(method, params))
+
+const getObject = (id: any) =>
+  xoCall('xo.getAllObjects', { filter: { id } }).then(objects => objects[id])
+
+export default class Visualization extends Component<any, any> {
+  state: any = {
+    granularity: 'seconds',
+    format: 'LTS',
+
+    //  Host
+    //  cpu
+    cpuDataHost: [],
+    cpusHost: [],
+
+    // Memory
+    memoryDataHost: [],
+    memoryMaxHost: 0,
+
+    //  Network
+    networkDataHost: [],
+    networksRxHost: [],
+    networksTransmissionHost: [],
+    networksReceptionHost: [],
+
+    //  Load
+    loadDataHost: [],
+    maxLoad: 0,
+  }
+
+  componentDidMount() {
+    setInterval(this.fetchHostStats.bind(this), 5e3)
+  }
+
+  fetchHostStats = () => {
+    getObject('470b64a4-2767-4e1d-a20c-fbc2a6d4de57').then((host: any) => {
+      this.setState({ memoryMaxHost: host.memory.size })
+    })
+   
+    xoCall('host.stats', {
+      host: '470b64a4-2767-4e1d-a20c-fbc2a6d4de57',
+      granularity: this.state.granularity,
+    }).then(
+      ({
+        endTimestamp,
+        stats: { cpus },
+        interval,
+        stats: { load },
+        stats: { memory, memoryFree = memory },
+        stats: { pifs },
+      }) => {
+        let format: any
+        if (interval === 5 || interval === 60) {
+          format = 'LTS'
+        } else if (interval === 86400 || interval === 3600) {
+          format = 'l'
+        }
+
+        this.setState({ cpusHost: Object.keys(cpus) })
+        this.setState({ networksRxHost: Object.keys(pifs.rx) })
+        this.setState({ networksTransmissionHost: Object.keys(pifs.tx) })
+
+        let cpuDataHost: any[] = []
+        let memoryDataHost: any[] = []
+        let networkDataHost: any[] = []
+        let loadDataHost: any[] = []
+        let newDataMemory: any = []
+  
+          newDataMemory = memoryFree.map(
+            (value: any, index: any) => memory[index] - value
+          )
+        
+        for (var i = 0; i < NB_VALUES; i++) {
+          let valuesHost: any = {}
+          let valuesHostMemory: any = {}
+          let valuesHostNetwork: any = {}
+          let valuesLoad: any = {}
+
+          valuesHost.time = moment(
+            (endTimestamp - (NB_VALUES - i - 1) * interval) * 1000
+          ).format(format)
+
+          this.state.cpusHost.forEach((property: string | number) => {
+            valuesHost[`cpu${property}`] = cpus[property][i]
+          })
+
+          this.state.networksRxHost.forEach((property: string | number) => {
+            valuesHostNetwork[`pifs_${property}_(rx)`] = pifs.rx[property][i]
+          })
+
+          this.state.networksTransmissionHost.forEach((property: string | number) => {
+            valuesHostNetwork[`pifs_${property}_(tx)`] = pifs.tx[property][i]
+          })
+
+          valuesLoad.load = load[i]
+          valuesHostMemory.memory = newDataMemory[i]
```

Comment:
If you don't need to change anything about those values, why don't you assign the whole array directly instead of copying each value one by one?